app/services/oa_workflow_service.rb

```diff
@@ -33,5 +33,14 @@ def workflow
     rescue StandardError
       file.update_column(:downloaded, false)
     end
+
+    ActivityInsightOAFile.needs_version_check.each do |file|
+      file.version_checked = true
+      file.save!
+      AiOAWfVersionCheckJob.perform_later(file.id)
+    rescue StandardError
+      file.update_column(:version_checked, false)
+      raise
+    end
```

Comment:
I think what we're doing here is ok since we're bypassing ActiveRecord and also re-raising the original error.